web/src/features/charts/bar-breakdown/utils.ts

```diff
@@ -192,3 +193,60 @@ export const getExchangesToDisplay = (
     (exchangeZoneKey) => !exchangeZoneKeysToRemove.has(exchangeZoneKey)
   );
 };
+
+export function getEmissionData(zoneData: ZoneDetails) {
+  const sourceInfoToProductionSource = new Map<string, string[]>();
+
+  for (const state of Object.values(zoneData.zoneStates)) {
+    updateMapWithSources(
+      state.dischargeCo2IntensitySources,
+      sourceInfoToProductionSource,
+      true
+    );
+    updateMapWithSources(
+      state.productionCo2IntensitySources,
+      sourceInfoToProductionSource
+    );
+  }
+
+  return sourceInfoToProductionSource;
+}
+
+function updateMapWithSources(
+  sources: { [key: string]: string },
+  sourceInfoToProductionSource: Map<string, string[]>,
+  storageType?: boolean
+) {
+  for (const entry of Object.entries(sources)) {
+    for (const sourceInfo of entry[1].split('; ')) {
+      const productionSource = getProductionSourcesToAdd(
+        entry,
+        sourceInfoToProductionSource.get(sourceInfo),
+        storageType
+      );
+      if (productionSource.length > 0) {
+        sourceInfoToProductionSource.set(sourceInfo, productionSource);
+      }
+    }
+  }
+}
+
+function getProductionSourcesToAdd(
+  entry: [string, string],
+  productionSourceArray: string[] | undefined,
+  storageType?: boolean
+): string[] {
+  const productionSource = storageType ? entry[0] + ' storage' : entry[0];
```

Comment:
Would really like a
```js
`${}`
```
here instead. Actually somewhat surprised we didn't have a ESLint rule that covered using the addition operator on strings.